pkgs/by-name/rq/rquickshare/package.nix

```
@@ -1,34 +1,118 @@
+      arg2
+    else
+      throw "Wrong argument for app-type in rquickshare package";
+
+  proper-cargo-tauri = app-type-either cargo-tauri cargo-tauri_1;
+in
+rustPlatform.buildRustPackage rec {
+  pname = "rquickshare" + (app-type-either "" "-legacy");
   version = "0.11.2";
-  src = fetchurl {
-    url = "https://github.com/Martichou/rquickshare/releases/download/v${version}/r-quick-share-legacy_v${version}_glibc-2.31_amd64.AppImage";
-    hash = "sha256-VXYiYrTSedH8xFjuxbdplzVdfnO6s3ftY2I121Unlfw=";
+
+  src = fetchFromGitHub {
+    owner = "Martichou";
+    repo = "rquickshare";
+    tag = "v${version}";
+    hash = "sha256-MYhM6FB3zfvMFnLMecbn4Ismvk8AAykIvhPBnFcNjdI=";
   };
-  appimageContents = appimageTools.extractType2 { inherit pname version src; };
-in
-appimageTools.wrapType2 {
-  inherit pname version src;
-  extraInstallCommands = ''
-    install -Dm444 ${appimageContents}/r-quick-share.desktop -t $out/share/applications
-    substituteInPlace $out/share/applications/r-quick-share.desktop \
-      --replace-fail 'Exec=r-quick-share' 'Exec=r-quick-share %u'
-    cp -r ${appimageContents}/usr/share/icons $out/share
+
+  # from https://github.com/NixOS/nixpkgs/blob/04e40bca2a68d7ca85f1c47f00598abb062a8b12/pkgs/by-name/ca/cargo-tauri/test-app.nix#L23-L26
+  postPatch = lib.optionalString stdenv.hostPlatform.isLinux ''
+    substituteInPlace $cargoDepsCopy/libappindicator-sys-*/src/lib.rs \
+      --replace-fail "libayatana-appindicator3.so.1" "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1"
   '';
 
+  pnpmRoot = "app/${app-type}";
+  pnpmDeps = pnpm.fetchDeps {
+    inherit pname version src;
+
+    sourceRoot = "${src.name}/app/${app-type}";
+    hash = app-type-either "sha256-VPb2idEiBruVCwz5lDojFmD2C5CfXYSe5mPWBwCGeJs=" "sha256-sDHysaKMdNcbL1szww7/wg0bGHOnEKsKoySZJJCcPik=";
+  };
+
+  useFetchCargoVendor = true;
+  cargoRoot = "app/${app-type}/src-tauri";
+  buildAndTestSubdir = cargoRoot;
+  cargoHash = app-type-either "sha256-5vGXMHar9E25jVseXR2uI82DtZXyPaHgDD8QkqoRhhY=" "sha256-kkbDEV4k+4EiUi4+EsZdmgm9/vMC0m5L5lpyCn7Zzb0=";
+
+  nativeBuildInputs = [
+    proper-cargo-tauri.hook
+
+    # Setup pnpm
+    nodejs
+    pnpm.configHook
+
+    # Make sure we can find our libraries
+    perl
+    pkg-config
+    protobuf
+    wrapGAppsHook4
+  ];
+
+  buildInputs =
+    [ openssl ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      glib-networking # Most Tauri apps need networking
+      libayatana-appindicator
+    ]
+    ++ lib.optionals (app-type == "main") [
+      webkitgtk_4_1
+      libsoup_3
+    ]
+    ++ lib.optionals (app-type == "legacy") [
+      webkitgtk_4_0
+      libsoup_2_4
+    ];
+
+  passthru.updateScript = lib.mkIf (app-type == "main") (
+    writeShellScript "update-rquickshare.sh" ''
+      ${lib.getExe nix-update} rquickshare
+      sed -i 's/version = "0.0.0";/' pkgs/by-name/rq/rquickshare/package.nix
+      ${lib.getExe nix-update} rquickshare-legacy
+    ''
+  );
```

Actually this will not work, my bad (`lib.mkIf` is only for the module system).

```suggestion
    if (app-type == "main") then
      (writeShellScript "update-rquickshare.sh" ''
        ${lib.getExe nix-update} rquickshare
        sed -i 's/version = "0.0.0";/' pkgs/by-name/rq/rquickshare/package.nix
        ${lib.getExe nix-update} rquickshare-legacy
      '')
    else
      null;
```